lib/graphql/type/schema.ex

```diff
@@ -81,6 +82,19 @@ defmodule GraphQL.Schema do
     end
   end
 
+  defp reduce_types(typemap, %Input{} = type) do
+    if Map.has_key?(typemap, type.name) do
+      typemap
+    else
+      typemap = Map.put(typemap, type.name, type)
+      thunk_fields = CompositeType.get_fields(type)
+      typemap = Enum.reduce(thunk_fields, typemap, fn({_,fieldtype},typemap) ->
```

Comment:
Should we start using Elixir 1.3 conventions and use the assignment form of the `if` expression?

```
typemap = if .... do
end
```

?